service/src/main/java/bio/terra/catalog/datarepo/DatarepoService.java

```diff
@@ -47,11 +51,21 @@ public DatarepoService(DatarepoConfiguration datarepoConfig) {
     this.commonHttpClient = new ApiClient().getHttpClient();
   }
 
-  public Map<String, List<String>> getSnapshotIdsAndRoles(AuthenticatedUserRequest user) {
+  public Map<String, List<DatasetAccessLevel>> getSnapshotIdsAndRoles(
```

Comment:
Do we still need `List` here?
```suggestion
  public Map<String, DatasetAccessLeve>> getSnapshotIdsAndRoles(
```

Reply:
After seeing your comment above:

> One thing that is weird is that we are going to return a list of Data Catalog roles rn in the form of a role array. I intend to fix this, but that one would be the breaking change, so I am going to spin it out again and do that as a follow on PR to make that change as small as possible.

This change could be done later too. And/or do the mapping from X -> List<X> in the caller instead of in the services.